core/src/main/java/cz/o2/proxima/transaction/KeyAttribute.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2017-2021 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.transaction;
+
+import com.google.common.base.Preconditions;
+import cz.o2.proxima.annotations.Experimental;
+import cz.o2.proxima.repository.AttributeDescriptor;
+import cz.o2.proxima.repository.EntityDescriptor;
+import java.io.Serializable;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+/**
+ * A combination of key of an entity, attribute descriptor and (optional) specific attribute. The
+ * specific attribute is needed when this object needs to describe a specific attribute of wildcard
+ * attribute descriptor.
+ */
+@Experimental
+@ToString
+@EqualsAndHashCode
+public class KeyAttribute implements Serializable {
```

Comment:
do we need this to be serializable? we should avoid java serialization where possible

Reply:
Unfortunately we need it to be serializable, because we fall back to Java Serialization when there is no other option how to serialize data between client and transaction manager - one can (and should!) define a provider for serialization scheme via
```
transactions {
  scheme-provider: proto
}
```
and it will use protobuffers then (needs `proxima-scheme-proto` dependency). If there is no explicit provider (and associted dependency), then default of Java serialization is used. This produces warnings.